src/ui/src/builder/settings/BuilderSettingsDeploySharedBlueprint.vue

```diff
@@ -0,0 +1,276 @@
+			<div
+				v-if="currentVersion"
+				class="DeploySharedBlueprint__versionInfo"
+			>
+				<WdsIcon name="package" />
+				<span
+					>Currently published:
+					<strong>v{{ currentVersion }}</strong></span
+				>
+			</div>
+			<div v-else class="DeploySharedBlueprint__versionInfo">
+				<WdsIcon name="package" />
+				<span>Not yet published</span>
+			</div>
+
+			<WdsFieldWrapper label="Name" required>
+				<WdsTextInput
+					v-model="form.name"
+					placeholder="e.g., Data Fetch Pipeline"
+					:error="errors.name"
+					@update:model-value="handleNameChange"
+				/>
+			</WdsFieldWrapper>
+
+			<WdsFieldWrapper label="Description" required>
+				<textarea
+					v-model="form.description"
+					placeholder="Describe what this shared blueprint does"
+					class="DeploySharedBlueprint__textarea"
+					:class="{
+						'DeploySharedBlueprint__textarea--error':
+							errors.description,
+					}"
+				/>
+			</WdsFieldWrapper>
+		</div>
+	</WdsModal>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, watch, inject } from "vue";
+import WdsModal, { ModalAction } from "@/wds/WdsModal.vue";
+import WdsFieldWrapper from "@/wds/WdsFieldWrapper.vue";
+import WdsTextInput from "@/wds/WdsTextInput.vue";
+import WdsIcon from "@/wds/WdsIcon.vue";
+import injectionKeys from "@/injectionKeys";
+import { useToasts } from "@/builder/useToast";
+import { useComponentActions } from "@/builder/useComponentActions";
+import { useWriterTracking } from "@/composables/useWriterTracking";
+import { useWriterApi } from "@/composables/useWriterApi";
+import { DEFAULT_ORG_ID } from "@/constants/sharedBlueprints";
+
+const wf = inject(injectionKeys.core);
+const wfbm = inject(injectionKeys.builderManager);
+const { pushToast } = useToasts();
+const { setContentValue, extractBlueprintComponents } = useComponentActions(
+	wf,
+	wfbm,
+);
+const tracking = useWriterTracking(wf);
+const { writerApi } = useWriterApi();
+
+const props = defineProps<{
+	modelValue: boolean;
+	blueprintId: string;
+}>();
+
+const emit = defineEmits<{
+	"update:modelValue": [value: boolean];
+}>();
+
+const isOpen = computed({
+	get: () => props.modelValue,
+	set: (value) => emit("update:modelValue", value),
+});
+
+const blueprint = computed(() => wf.getComponentById(props.blueprintId));
+
+const currentVersion = computed(
+	() => blueprint.value?.content?.deployedVersion || null,
+);
+
+const blueprintName = computed(
+	() => blueprint.value?.content?.key || "Shared Blueprint",
+);
+
+const blueprintDescription = computed(
+	() => blueprint.value?.content?.deployedDescription || "",
+);
+
+const form = ref({
+	name: "",
+	description: "",
+});
+
+const errors = ref<Record<string, string>>({});
+const isDeploying = ref(false);
+
+function validateForm(): boolean {
+	errors.value = {};
+
+	if (!form.value.name.trim()) {
+		errors.value.name = "Name is required";
+	}
+	if (!form.value.description.trim()) {
+		errors.value.description = "Description is required";
+	}
```

Comment:
It's better to assign the object to avoid surprises

```suggestion
const errors = shallowRef<Record<string, string>>({});
const isDeploying = ref(false);

function validateForm(): boolean {
	const newErrors = {};

	if (!form.value.name.trim()) {
		newErrors.name = "Name is required";
	}
	if (!form.value.description.trim()) {
		newErrors.description = "Description is required";
	}
	errors.value = newErrors;
```